File: frontend/src/shared/data/colombiaDepartamentosCiudades.ts
/**
 * Catálogo de Departamentos y Ciudades (municipios principales) de Colombia.
 * Fuente de verdad para el frontend web. Existe un equivalente en Dart para
 * la app móvil (movil/lib/core/colombia_departamentos_ciudades.dart) — si se
 * agrega o corrige una ciudad aquí, replicar el cambio también allá.
 */

export interface DepartamentoCiudades {
  departamento: string;
  ciudades: string[];
}

export const COLOMBIA_DEPARTAMENTOS_CIUDADES: DepartamentoCiudades[] = [
  { departamento: 'Amazonas', ciudades: ['Leticia', 'Puerto Nariño'] },
  {
    departamento: 'Antioquia',
    ciudades: [
      'Apartadó', 'Bello', 'Caldas', 'Caucasia', 'Chigorodó', 'Copacabana',
      'El Carmen de Viboral', 'Envigado', 'Girardota', 'Itagüí', 'La Estrella',
      'Marinilla', 'Medellín', 'Necoclí', 'Rionegro', 'Sabaneta',
      'Santa Fe de Antioquia', 'Turbo',
    ],
  },
  { departamento: 'Arauca', ciudades: ['Arauca', 'Arauquita', 'Saravena', 'Tame'] },
  {
    departamento: 'Atlántico',
    ciudades: ['Baranoa', 'Barranquilla', 'Galapa', 'Malambo', 'Puerto Colombia', 'Sabanalarga', 'Soledad'],
  },
  {
    departamento: 'Bogotá D.C.',
    ciudades: ['Bogotá'],
  },
  {
    departamento: 'Bolívar',
    ciudades: ['Arjona', 'Cartagena', 'El Carmen de Bolívar', 'Magangué', 'Mompós', 'Turbaco'],
  },
  {
    departamento: 'Boyacá',
    ciudades: ['Chiquinquirá', 'Duitama', 'Paipa', 'Sogamoso', 'Tunja', 'Villa de Leyva'],
  },
  {
    departamento: 'Caldas',
    ciudades: ['Anserma', 'Chinchiná', 'La Dorada', 'Manizales', 'Riosucio', 'Villamaría'],
  },
  { departamento: 'Caquetá', ciudades: ['Florencia', 'Puerto Rico', 'San Vicente del Caguán'] },
  { departamento: 'Casanare', ciudades: ['Aguazul', 'Tauramena', 'Villanueva', 'Yopal'] },
  { departamento: 'Cauca', ciudades: ['Patía', 'Popayán', 'Puerto Tejada', 'Santander de Quilichao'] },
  { departamento: 'Cesar', ciudades: ['Aguachica', 'Bosconia', 'Codazzi', 'Valledupar'] },
  { departamento: 'Chocó', ciudades: ['Istmina', 'Quibdó', 'Tadó'] },
  { departamento: 'Córdoba', ciudades: ['Cereté', 'Lorica', 'Montería', 'Sahagún', 'Tierralta'] },
  {
    departamento: 'Cundinamarca',
    ciudades: [
      'Cajicá', 'Chía', 'Facatativá', 'Funza', 'Fusagasugá', 'Girardot',
      'Madrid', 'Mosquera', 'Soacha', 'Zipaquirá',
    ],
  },
  { departamento: 'Guainía', ciudades: ['Inírida'] },
  { departamento: 'Guaviare', ciudades: ['San José del Guaviare'] },
  { departamento: 'Huila', ciudades: ['Garzón', 'La Plata', 'Neiva', 'Pitalito'] },
  { departamento: 'La Guajira', ciudades: ['Maicao', 'Manaure', 'Riohacha', 'Uribia'] },
  { departamento: 'Magdalena', ciudades: ['Ciénaga', 'El Banco', 'Fundación', 'Santa Marta'] },
  { departamento: 'Meta', ciudades: ['Acacías', 'Granada', 'Puerto López', 'Villavicencio'] },
  { departamento: 'Nariño', ciudades: ['Ipiales', 'Pasto', 'Tumaco', 'Túquerres'] },
  {
    departamento: 'Norte de Santander',
    ciudades: ['Cúcuta', 'Los Patios', 'Ocaña', 'Pamplona', 'Villa del Rosario'],
  },
  { departamento: 'Putumayo', ciudades: ['Mocoa', 'Orito', 'Puerto Asís'] },
  { departamento: 'Quindío', ciudades: ['Armenia', 'Calarcá', 'La Tebaida', 'Montenegro'] },
  {
    departamento: 'Risaralda',
    ciudades: ['Dosquebradas', 'La Virginia', 'Pereira', 'Santa Rosa de Cabal'],
  },
  { departamento: 'San Andrés y Providencia', ciudades: ['Providencia', 'San Andrés'] },
  {
    departamento: 'Santander',
    ciudades: [
      'Barrancabermeja', 'Bucaramanga', 'Floridablanca', 'Girón',
      'Piedecuesta', 'San Gil',
    ],
  },
  { departamento: 'Sucre', ciudades: ['Corozal', 'Sampués', 'Sincelejo'] },
  { departamento: 'Tolima', ciudades: ['Espinal', 'Honda', 'Ibagué', 'Melgar'] },
  {
    departamento: 'Valle del Cauca',
    ciudades: ['Buenaventura', 'Buga', 'Cali', 'Cartago', 'Jamundí', 'Palmira', 'Tuluá', 'Yumbo'],
  },
  { departamento: 'Vaupés', ciudades: ['Mitú'] },
  { departamento: 'Vichada', ciudades: ['Puerto Carreño'] },
];

export function getAllDepartamentos(): string[] {
  return COLOMBIA_DEPARTAMENTOS_CIUDADES.map(d => d.departamento);
}

export function getCiudadesByDepartamento(departamento: string): string[] {
  return COLOMBIA_DEPARTAMENTOS_CIUDADES.find(d => d.departamento === departamento)?.ciudades ?? [];
}

File: movil/lib/core/colombia_departamentos_ciudades.dart
// Catálogo de Departamentos y Ciudades (municipios principales) de Colombia.
// Equivalente Dart del catálogo web en
// frontend/src/shared/data/colombiaDepartamentosCiudades.ts — si se agrega
// o corrige una ciudad aquí, replicar el cambio también allá.

class DepartamentoCiudades {
  final String departamento;
  final List<String> ciudades;
  const DepartamentoCiudades(this.departamento, this.ciudades);
}

const List<DepartamentoCiudades> kColombiaDepartamentosCiudades = [
  DepartamentoCiudades('Amazonas', ['Leticia', 'Puerto Nariño']),
  DepartamentoCiudades('Antioquia', [
    'Apartadó', 'Bello', 'Caldas', 'Caucasia', 'Chigorodó', 'Copacabana',
    'El Carmen de Viboral', 'Envigado', 'Girardota', 'Itagüí', 'La Estrella',
    'Marinilla', 'Medellín', 'Necoclí', 'Rionegro', 'Sabaneta',
    'Santa Fe de Antioquia', 'Turbo',
  ]),
  DepartamentoCiudades('Arauca', ['Arauca', 'Arauquita', 'Saravena', 'Tame']),
  DepartamentoCiudades('Atlántico', [
    'Baranoa', 'Barranquilla', 'Galapa', 'Malambo', 'Puerto Colombia',
    'Sabanalarga', 'Soledad',
  ]),
  DepartamentoCiudades('Bogotá D.C.', ['Bogotá']),
  DepartamentoCiudades('Bolívar', [
    'Arjona', 'Cartagena', 'El Carmen de Bolívar', 'Magangué', 'Mompós', 'Turbaco',
  ]),
  DepartamentoCiudades('Boyacá', [
    'Chiquinquirá', 'Duitama', 'Paipa', 'Sogamoso', 'Tunja', 'Villa de Leyva',
  ]),
  DepartamentoCiudades('Caldas', [
    'Anserma', 'Chinchiná', 'La Dorada', 'Manizales', 'Riosucio', 'Villamaría',
  ]),
  DepartamentoCiudades('Caquetá', ['Florencia', 'Puerto Rico', 'San Vicente del Caguán']),
  DepartamentoCiudades('Casanare', ['Aguazul', 'Tauramena', 'Villanueva', 'Yopal']),
  DepartamentoCiudades('Cauca', ['Patía', 'Popayán', 'Puerto Tejada', 'Santander de Quilichao']),
  DepartamentoCiudades('Cesar', ['Aguachica', 'Bosconia', 'Codazzi', 'Valledupar']),
  DepartamentoCiudades('Chocó', ['Istmina', 'Quibdó', 'Tadó']),
  DepartamentoCiudades('Córdoba', ['Cereté', 'Lorica', 'Montería', 'Sahagún', 'Tierralta']),
  DepartamentoCiudades('Cundinamarca', [
    'Cajicá', 'Chía', 'Facatativá', 'Funza', 'Fusagasugá', 'Girardot',
    'Madrid', 'Mosquera', 'Soacha', 'Zipaquirá',
  ]),
  DepartamentoCiudades('Guainía', ['Inírida']),
  DepartamentoCiudades('Guaviare', ['San José del Guaviare']),
  DepartamentoCiudades('Huila', ['Garzón', 'La Plata', 'Neiva', 'Pitalito']),
  DepartamentoCiudades('La Guajira', ['Maicao', 'Manaure', 'Riohacha', 'Uribia']),
  DepartamentoCiudades('Magdalena', ['Ciénaga', 'El Banco', 'Fundación', 'Santa Marta']),
  DepartamentoCiudades('Meta', ['Acacías', 'Granada', 'Puerto López', 'Villavicencio']),
  DepartamentoCiudades('Nariño', ['Ipiales', 'Pasto', 'Tumaco', 'Túquerres']),
  DepartamentoCiudades('Norte de Santander', [
    'Cúcuta', 'Los Patios', 'Ocaña', 'Pamplona', 'Villa del Rosario',
  ]),
  DepartamentoCiudades('Putumayo', ['Mocoa', 'Orito', 'Puerto Asís']),
  DepartamentoCiudades('Quindío', ['Armenia', 'Calarcá', 'La Tebaida', 'Montenegro']),
  DepartamentoCiudades('Risaralda', [
    'Dosquebradas', 'La Virginia', 'Pereira', 'Santa Rosa de Cabal',
  ]),
  DepartamentoCiudades('San Andrés y Providencia', ['Providencia', 'San Andrés']),
  DepartamentoCiudades('Santander', [
    'Barrancabermeja', 'Bucaramanga', 'Floridablanca', 'Girón',
    'Piedecuesta', 'San Gil',
  ]),
  DepartamentoCiudades('Sucre', ['Corozal', 'Sampués', 'Sincelejo']),
  DepartamentoCiudades('Tolima', ['Espinal', 'Honda', 'Ibagué', 'Melgar']),
  DepartamentoCiudades('Valle del Cauca', [
    'Buenaventura', 'Buga', 'Cali', 'Cartago', 'Jamundí', 'Palmira', 'Tuluá', 'Yumbo',
  ]),
  DepartamentoCiudades('Vaupés', ['Mitú']),
  DepartamentoCiudades('Vichada', ['Puerto Carreño']),
];

List<String> todosLosDepartamentos() =>
    kColombiaDepartamentosCiudades.map((d) => d.departamento).toList();

List<String> ciudadesPorDepartamento(String departamento) {
  for (final d in kColombiaDepartamentosCiudades) {
    if (d.departamento == departamento) return d.ciudades;
  }
  return const [];
}
